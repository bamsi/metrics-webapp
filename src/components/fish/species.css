.list-container {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  color: #fffff7b0;
}

.background {
  width: 100%;
  height: 100%;
}

.background img {
  width: 100%;
  height: 100%;
  object-fit: cover;
  opacity: 60%;
}

.list-title {
  display: block;
  font-weight: 700;
  font-size: 1.2rem;
  color: #fffff7b0;
  padding-bottom: 0.5rem;
}

.content-section {
  padding: 1rem;
}

.list-item {
  display: grid;
  grid-template-rows: 3fr 1fr;
  margin: 0.2rem;
  background-color: #31347354;
  padding: 0.7rem;
  position: relative;
  cursor: pointer;
}

.description {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-end;
}

.description .title {
  font-weight: 500;
  font-size: 1rem;
  color: #ffff;
}

.description .sub-title {
  font-weight: 400;
  font-size: 0.8rem;
}

.arrow {
  position: absolute;
  top: 1rem;
  right: 1rem;
  z-index: 10;
}

.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-bottom: 0.5rem;
}

.search-box {
  border-radius: 0.5rem;
  border: 1px solid #31347354;
  padding: 0.3rem;
}

.search-box::placeholder {
  font-style: italic;
}

.search-button {
  border-radius: 0.5rem;
  border: 1px solid #000;
  padding: 0.3rem;
  margin-left: 0.2rem;
}

.not-found {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  color: #fff;
}

@media only screen and (min-width: 760px) {
  .content-section {
    margin: 0 10%;
  }

  .list-container {
    grid-template-columns: repeat(3, 1fr);
  }

  .search-box {
    width: 17rem;
  }

  .search-button {
    width: 5rem;
  }
}
