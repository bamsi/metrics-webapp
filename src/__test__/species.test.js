import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Species from '../components/fish/species';

describe('Testing rendering of the mission page', () => {
  it('', () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Species />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
