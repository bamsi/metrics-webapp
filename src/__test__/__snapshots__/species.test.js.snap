// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing rendering of the mission page  1`] = `
<div>
  <div
    class="content-section"
  >
    <header
      class="header"
    >
      <h2
        class="list-title"
      >
        Fish Species
      </h2>
      <form>
        <input
          class="search-box"
          placeholder="Search by species"
          type="text"
          value=""
        />
        <input
          class="search-button"
          type="submit"
          value="Search"
        />
      </form>
    </header>
    <div
      class="not-found"
    >
       No data found!
    </div>
  </div>
</div>
`;
